[cleanup] Move _redirects from public to static

chalin opened the pull request:
- Fixes #43
  - Moves `_redirects` from `public` to `static`
- Adjusts `layouts/partials/css.html` so that it creates a production build only in production (when `hugo.IsProduction` is true)
- Adds NPM scripts to build and serve the website. Note that these scripts aren't used by the Netlify build commands yet.

### Redirect tests

- https://deploy-preview-51--in-toto.netlify.app/Statement/v0.1.0
- https://deploy-preview-51--in-toto.netlify.app/attestation/vulns/v...

chalin commented:
/cc @marcelamelara @steiza - FYI, so that you're aware that the canonical source of the _redirects file has changed.

chalin commented:
@DarikshaAnsari - we'll need to ensure that the `_redirects` file is also present in the Docsy-based version of the website.

lukpueh reviewed (commented):
>   * Moves `_redirects` from `public` to `static`
👍 

> * Adjusts `layouts/partials/css.html` so that it creates a production build only in production (when `hugo.IsProduction` is true)

TBH, I can't tell if this is correct. Should I see a difference in the preview build? It looks like it does use the production css link (w/ hash):
```
<link rel="stylesheet" href="/css/style.94f9acd6bd00126cfa5eeb849dbf3cc631715b317a956d7516a0d3e4e1c1cef7.css" integrity="sha256-lPms1r0AEmz6XuuEnb88xjFxWzF6lW11FqDT5OHBzvc=">
```

> * Adds NPM scripts to build and serve the website. Note that these scripts aren't used by the Netlify build commands yet.

Are these commands also for devs to build locally? If so, should we update our CONTRIBUTING.md? It does not have to be part of this pr.

> * https://deploy-preview-51--in-toto.netlify.app/Statement/v0.1.0
> * https://deploy-preview-51--in-toto.netlify.app/attestation/vulns/v...

💯 for providing links here.

chalin commented:
> > * Adjusts `layouts/partials/css.html` so that it creates a production build only in production (when `hugo.IsProduction` is true)
> 
> TBH, I can't tell if this is correct. Should I see a difference in the preview build? It looks like it does use the production css link (w/ hash):

Right, I had not enabled the scripts for use by Netlify. I've added that change so that you can see the effect by viewing source of the preview. Let me know what you think.

> > * Adds NPM scripts to build and serve the website. Note that these scripts aren't used by the Netlify build commands yet.
> 
> Are these commands also for devs to build locally? If so, should we update our CONTRIBUTING.md? It does not have to be part of this pr.

Both for use by Netlify and devs. I've updated `CONTRIBUTING.md` too.

> > * https://deploy-preview-51--in-toto.netlify.app/Statement/v0.1.0
> > * https://deploy-preview-51--in-toto.netlify.app/attestation/vulns/v...
> 
> 💯 for providing links here.

👍🏻

lukpueh commented on package.json:
```diff
@@ -1,6 +1,20 @@
+    "build:production": "npm run _hugo -- --minify",
+    "build": "npm run _build --",
+    "clean": "rm -Rf node_modules public/* resources",
+    "make:public": "mkdir -p public && git init -b main public",
```
What's this command needed for?

chalin replied:
It creates the `public` folder and sets it up as a local git repo. This makes it easier to view generated file diff when working on website changes. It's more of a power-user feature. Would you like for me to remove it?

lukpueh replied:
No, it's fine.

lukpueh commented on CONTRIBUTING.md:
````diff
@@ -35,17 +35,17 @@ npm install
 
 ```sh
-make serve
+npm run serve
````
Did you try this? It fails on my machine with this error:

```
> hugo --cleanDestinationDir -e dev -DF --minify --disableFastRender serve

Error: unknown flag: --disableFastRender
```

Comparing this to the removed Makefile, it looks like it's missing the `server` subcommand.

chalin replied:
Fixed. No, it was because of the wrong order of arguments.

lukpueh commented:
> > TBH, I can't tell if this is correct. Should I see a difference in the preview build? It looks like it does use the production css link (w/ hash):
> 
> Right, I had not enabled the scripts for use by Netlify. I've added that change so that you can see the effect by viewing source of the preview. Let me know what you think.

Looks like expected now. Thanks!
